app/controllers/qsets_controller.rb

```diff
@@ -10,6 +10,36 @@ def index
     end
   end
 
+  def qset_json
+    params.permit(:filter)
+    filter = params[:filter]
+    answers_json_data = []
+    $questionCount = 0
+    if filter == 'other'
+      @raw_questions = Question.includes(:answers).where(qset_id: @qset.id).where.not(user_id: current_user)
```

Comment:
I think this json response was missing an extra piece -- the permissions on the qset. I've incorporated that into the suggested refactored code.